src/ORM/DataList.php

```diff
@@ -1079,173 +1039,205 @@ private function executeQuery(): Query
 
     private function fetchEagerLoadRelations(Query $query): void
```

Comment:
This method has been refactored so that:
1. Instead of having a full list of _partial_ relations in the chain (e.g. `relation1`, `relation1.relation2`, etc) it now only contains the _complete_ chains (e.g. `relation1.relation2`).
2. It loops over each relation in the chain, and _always_ passes the parent relation data into the various `fetchEagerLoad*()` methods
3. switch/case is used instead of chained if statements, as it is clearer to read

In turn, each of the `fetchEagerLoad*()` methods has been changed so that once they have fetched their data, they add it to the appropriate parent. For top level relations that's into the `eagerLoadedData` array. For subrelations that's the appropriate parent `EagerLoadedDataList` or `DataObject` record.